AirCasting/Map/Heatmap.swift

```diff
@@ -0,0 +1,344 @@
+    }
+    
+    private func getGridSquare(x: Int, y: Int) -> GridSquare? {
+        return gridSquares[gridSquareKey(x, y)]
+    }
+    
+    private mutating func setGridSquare(x: Int, y: Int, gridSquare: GridSquare?) {
+        gridSquares[gridSquareKey(x, y)] = gridSquare
+    }
+    
+    private func getVisibleRegionCoordinates(mapView: GMSMapView) -> VisibleRegionCoordinates {
+        let visibleRegion = mapView.projection.visibleRegion()
+        
+        return VisibleRegionCoordinates(
+            visibleRegion.farLeft.latitude,
+            visibleRegion.nearLeft.latitude,
+            visibleRegion.farRight.longitude,
+            visibleRegion.nearLeft.longitude
+        )
+        
+    }
+    
+    private mutating func initGrid() {
+        
+        let lonGridSize = (visibleRegion.lonEast - visibleRegion.lonWest) / Double(gridSizeX)
+        let latGridSize = (visibleRegion.latNorth - visibleRegion.latSouth) / Double(gridSizeY)
+        
+        for x in 1...gridSizeX {
+            for y in 1...gridSizeY {
+                let gridCalculator = GridCalculator(x, y, visibleRegion, lonGridSize, latGridSize)
+                
+                setGridSquare(x, y, GridSquare(
+                    mapView: mapView,
+                    sensorThreshold: sensorThreshold,
+                    gridCalculator.southWestLatLng(),
+                    gridCalculator.southEastLatLng(),
+                    gridCalculator.northEastLatLng(),
+                    gridCalculator.northWestLatLng()
+                ))
+            }
+        }
+    }
+    
+    private mutating func setGridSquare(_ x: Int, _ y: Int, _ gridSquare: GridSquare?) {
+        self.gridSquares[gridSquareKey(x, y)] = gridSquare
+    }
+    
+    private func gridSquareKey(_ x: Int, _ y: Int) -> String {
+        return "\(x)_\(y)"
+    }
+    
+    struct GridCalculator {
+        let x: Int
+        let y: Int
+        let visibleRegion: VisibleRegionCoordinates
+        let lonGridSize: Double
+        let latGridSize: Double
+        
+        init(_ x: Int, _ y: Int, _ visibleRegion: VisibleRegionCoordinates, _ lonGridSize: Double, _ latGridSize: Double) {
+            self.x = x
+            self.y = y
+            self.visibleRegion = visibleRegion
+            self.lonGridSize = lonGridSize
+            self.latGridSize = latGridSize
+        }
+        
+        func southWestLatLng() -> CLLocationCoordinate2D {
+            return CLLocationCoordinate2D(latitude: southLatitude(), longitude: westLongitude())
+        }
+        
+        func southEastLatLng() -> CLLocationCoordinate2D {
+            return CLLocationCoordinate2D(latitude: southLatitude(), longitude: eastLongitude())
+        }
+        
+        func northEastLatLng() -> CLLocationCoordinate2D {
+            return CLLocationCoordinate2D(latitude: northLatitude(), longitude: eastLongitude())
+        }
+        
+        func northWestLatLng() -> CLLocationCoordinate2D {
```

Comment:
Can we move this whole struct to its own file. I think that creating a folder called Heatmap or something and placing there those structs would be great :)